test/common/matcher/exact_map_matcher_test.cc

```diff
@@ -14,9 +14,13 @@ namespace Matcher {
 struct TestData {};
 
 struct StringAction : public Action {
```

Comment:
nit: I think you could have an ActionBase or something that implements the typeUrl() stuff.